data/src/config/psql.py

```diff
@@ -1,12 +1,34 @@
 import os
 
-from config.config import is_docker
 from sqlalchemy import create_engine
 
-url: str = (
-    os.environ["VACANT_LOTS_DB"].replace("localhost", "host.docker.internal")
-    if is_docker()
-    else os.environ["VACANT_LOTS_DB"]
-)
+from config.config import is_docker
+
+# Detect if running in Cloud Run:
+is_cloud_run = "K_SERVICE" in os.environ or "CLOUD_RUN_JOB" in os.environ
+
+# Use host.docker.internal when running locally in Docker
+# except when running in Cloud Run
+if is_docker() and not is_cloud_run:
+    host = "host.docker.internal"
+else:
+    host = "localhost"
+
+
+if os.getenv("VACANT_LOTS_DB"):
+    # Use the provided database URL
+    url = os.getenv("VACANT_LOTS_DB")
```

Comment:
I probably should switch this back to 
```
 os.environ["VACANT_LOTS_DB"].replace("localhost", "host.docker.internal")
    if is_docker()
    else os.environ["VACANT_LOTS_DB"]
```
in-case we're in docker and VACANT_LOTS_DB is defined.